contentgrid-spring-audit-logging/src/test/java/com/contentgrid/spring/audit/AuditObservationHandlerTest.java

```diff
@@ -434,9 +437,13 @@ void contentDelete() throws Exception {
     @Nested
     class HandlesThrownExceptions {
 
-        private static final ServerRequestObservationContext SERVER_REQUEST_OBSERVATION_CONTEXT = new ServerRequestObservationContext(
-                new MockHttpServletRequest(),
-                new MockHttpServletResponse());
+        private static final ServerRequestObservationContext SERVER_REQUEST_OBSERVATION_CONTEXT = createContext();
+
+        private static ServerRequestObservationContext createContext() {
+            var request = new MockHttpServletRequest();
+            request.setAttribute(HandlerMapping.BEST_MATCHING_HANDLER_ATTRIBUTE, "true");
```

Comment:
why is this necessary ?

Reply:
The tests used to work because the `handler` in `BasicAuditEventExtractor.createEventBuilder` is `null` and it was missing a null check. Since the null-check is added and the requests are crafted manually, we add this attribute to make the handler non-null and the request isn't ignored.

Reply:
isn't it fishy that all the code expect that it contains a request-handler-method and we're putting the string `"true"` in there ? :thinking: